app/src/main/java/com/example/bookwormadventuresdeluxe2/MyProfileFragment.java

```diff
@@ -71,24 +105,57 @@ public void onClick(View view)
                 case R.id.profile_edit:
                     final View editInfo = LayoutInflater.from(this.getContext()).inflate(R.layout.edit_profile, null);
 
-                    // Set up the input
-                    final EditText inputEmail = editInfo.findViewById(R.id.edit_email);
-                    final EditText inputPhone = editInfo.findViewById(R.id.edit_phone);
-                    // Specify the type of input expected
+                    /* Set up the input */
+                    EditText inputEmail = editInfo.findViewById(R.id.edit_email);
+                    EditText inputPhone = editInfo.findViewById(R.id.edit_phone);
+
+                    /* Specify the type of input expected */
                     inputEmail.setInputType(InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS);
                     inputPhone.setInputType(InputType.TYPE_CLASS_PHONE);
 
+                    /* Setting text to user's details */
+                    inputEmail.setText(viewUserObject.getEmail());
+                    inputPhone.setText(viewUserObject.getPhoneNumber());
+
                     final AlertDialog builder = new AlertDialog.Builder(this.getContext()).create();
                     builder.setView(editInfo);
 
-                    // Set up the buttons
+                    /* Set up the buttons */
                     editInfo.findViewById(R.id.edit_confirm).setOnClickListener(new View.OnClickListener()
                     {
                         @Override
                         public void onClick(View view)
                         {
-                            //TODO: test input, get input, update user
-                            builder.dismiss();
+                            /* Checks if no changes were made */
+                            if (viewUserObject.getEmail().compareTo(inputEmail.getText().toString()) == 0
+                                && viewUserObject.getPhoneNumber().compareTo(inputPhone.getText().toString()) == 0)
+                            {
```

Comment:
could switch to getEmail().equals(inputEmail.getText.toString()) to avoid extra comparisons

Also is there a reason the button functions are in a switch case instead of just having separate listeners?

Reply:
i tried moving the popup fragment buttons but they require the initial profile_edit button call with them

Reply:
![image](https://user-images.githubusercontent.com/39696855/97214528-d915ae80-1788-11eb-8ae7-fc99d9b243ac.png)